src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,46 +1,64 @@
-     * (обе даты включительно) составленный по следующей форме: Отчёт за период
-     * #дата_1# - #дата_2# Имя сотрудника - сумма заработанных средств за этот период
-     * Создать класс-ошибку IllegalDateParametersException и сделать так, чтобы
-     * метод getSalaryInfo выбрасывал IllegalDateParametersException,
-     * если dateFrom > dateTo, с сообщнием "Wrong parameters"</p>
-     *
-     * <p>Пример ввода: date from = 01.04.2019 date to = 30.04.2019</p>
-     *
-     * <p>names:
-     * Сергей
-     * Андрей
-     * София</p>
-     *
-     * <p>data:
-     * 26.04.2019 Сергей 60 50
-     * 26.04.2019 Андрей 3 200
-     * 26.04.2019 Сергей 7 100
-     * 26.04.2019 София 9 100
-     * 26.04.2019 Сергей 11 50
-     * 26.04.2019 Андрей 3 200
-     * 26.04.2019 Сергей 7 100
-     * 26.04.2019 София 9 100
-     * 26.04.2019 Сергей 11 50</p>
-     *
-     * <p>Пример вывода:
-     * Отчёт за период 01.04.2019  - 30.04.2019
-     * Сергей - 1550
-     * Андрей - 600
-     * София - 900</p>
-     */
+
+    private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
-            throws Exception {
-        return null;
+            throws IllegalDateParametersException, ParseException {
+
+        Date startDate = DATE_FORMAT.parse(dateFrom);
+        Date endDate = DATE_FORMAT.parse(dateTo);
+
+        if (startDate.after(endDate)) {
+            throw new IllegalDateParametersException("Wrong parameters");
+        }
+
+        Map<String, Integer> salaryMap = new HashMap<>();
+        for (String name : names) {
+            salaryMap.put(name, 0);
+        }
+
+        for (String entry : data) {
+            String[] parts = entry.split(" ");
+            Date workDate = DATE_FORMAT.parse(parts[0]);
+            String employeeName = parts[1];
+            int hoursWorked = Integer.parseInt(parts[2]);
+            int hourlyRate = Integer.parseInt(parts[3]);
+
+            if (!workDate.before(startDate) && !workDate.after(endDate)) {
+                int salaryForDay = hoursWorked * hourlyRate;
+                salaryMap.put(employeeName, salaryMap.getOrDefault(employeeName, 0) + salaryForDay);
+            }
+        }
+
+        StringBuilder report = new StringBuilder("Отчёт за период ")
+                .append(dateFrom).append(" - ").append(dateTo).append("\n");
+
+        for (String name : names) {
+            report.append(name).append(" - ").append(salaryMap.get(name)).append("\n");
+        }
+
+        return report.toString().trim() + "\n";
+    }
+
+    public String generateReports(String[] names, String[] data, String[] dates)
+            throws IllegalDateParametersException, ParseException {
+        StringBuilder allReports = new StringBuilder();
+
+        for (int i = 0; i < dates.length; i++) {
+            String report = getSalaryInfo(names, data, dates[0], dates[i]);
```

Comment:
In the `generateReports` method, the loop uses `dates[0]` as the start date for all reports, which is likely incorrect. Each report should have a unique start and end date pair, probably `dates[i]` should be used for both start and end dates in each iteration to generate reports for each date range.